.travis.yml

```diff
@@ -14,11 +14,24 @@ script:
 ## run eslint, flow and the unit test suite.
 - COVERAGE=y NODE_ENV=production npm test
 
+## NOTE: by setting the configured python to /bin/false we are forcing the production mode tests
+## to fail if any of the dependencies is a binary dependency that is build using node-gyp.
+- npm config set python /bin/false
+
 ## run functional test suite in a npm production environment
 ## (See #1082 for rationale).
 - npm run copy-dist-files-to-artifacts-dir
-- cd artifacts/production && npm install --production && cd -
-- TEST_WEB_EXT_BIN=./artifacts/production/bin/web-ext npm run test:functional
+- cd artifacts/production
+- npm install --production; export NPM_INSTALL_PROD=$?
+- cd -
+- test $NPM_INSTALL_PROD -eq 0 &&
```

Comment:
This looks like it will do nothing if `npm install --production` fails, correct? Will the test still fail in that case?

Reply:
yes, when `npm install --production` fails, the travis job will still fail, at line 27 when we test the return value.

Here is a test on travis which show the failure:

- https://travis-ci.org/mozilla/web-ext/builds/313421003#L2903